WarOfEmpires.Domain.Tests/Players/PlayerTests.cs

```diff
@@ -546,126 +556,103 @@ public sealed class PlayerTests {
         }
 
         [TestMethod]
-        public void Player_TrainTroops_Trains_Troops() {
+        public void Player_TrainTroops_Trains_Troops_Existing_TroopType() {
             var player = new Player(0, "Test");
             typeof(Player).GetProperty(nameof(Player.Resources)).SetValue(player, new Resources(200000, 10000, 10000, 10000, 10000));
 
-            player.TrainTroops(1, 4, 2, 5, 3, 6);
-
-            player.Archers.Should().Be(new Troops(1, 4));
-            player.Cavalry.Should().Be(new Troops(2, 5));
-            player.Footmen.Should().Be(new Troops(3, 6));
-        }
-
-        [TestMethod]
-        public void Player_TrainTroops_Removes_Peasants() {
-            var player = new Player(0, "Test");
-            typeof(Player).GetProperty(nameof(Player.Resources)).SetValue(player, new Resources(200000, 10000, 10000, 10000, 10000));
+            player.Troops.Add(new Troops(TroopType.Cavalry, 1, 1));
 
-            player.TrainTroops(1, 4, 2, 5, 3, 6);
+            player.TrainTroops(TroopType.Cavalry, 1, 4);
 
-            player.Peasants.Should().Be(4);
+            player.Troops.Should().HaveCount(1);
+            player.Troops.Single().Soldiers.Should().Be(2);
+            player.Troops.Single().Mercenaries.Should().Be(5);
         }
 
         [TestMethod]
-        public void Player_TrainTroops_Costs_Resources() {
+        public void Player_TrainTroops_Trains_Troops_New_TroopType() {
```

Comment:
Is the intent of the test just to test training of a troop type the player doesn't have, even if they don't have any troops previously, or should the test player have troops of a different type before the training?

Reply:
The former, but improved it a bit